src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1451,9 +1446,9 @@ abstract class RefChecks extends Transform {
             )
       }
 
-      sym.isSourceMethod &&
+      sym.name == nme.apply &&
+        !(sym hasFlag STABLE) && // ???
```

Comment:
The obvious question is: why not check `ACCESSOR` rather than `STABLE` if we want to "exclude all accessors"? 

I think the answer to this is that there are accessor-like methods that don't have the `ACCESSOR` flag set. One example I found is case field accessors:

```

scala> case class C(private val privParam: Int)
defined class C

scala> val caseFieldAccessor = typeOf[C].decl(TermName("privParam$1"))
caseFieldAccessor: $r.intp.global.Symbol = value privParam$1

scala> caseFieldAccessor.isStable
res22: Boolean = true

scala> caseFieldAccessor.isAccessor
res23: Boolean = false
```